perllib/FixMyStreet/Cobrand/Oxfordshire.pm

```diff
@@ -284,4 +289,108 @@ sub dashboard_export_problems_add_columns {
     });
 }
 
+sub defect_wfs_query {
+    my ($self, $bbox) = @_;
+
+    my $filter = "
+    <ogc:Filter xmlns:ogc=\"http://www.opengis.net/ogc\">
+        <ogc:And>
+            <ogc:PropertyIsEqualTo matchCase=\"true\">
+                <ogc:PropertyName>APPROVAL_STATUS_NAME</ogc:PropertyName>
+                <ogc:Literal>With Contractor</ogc:Literal>
+            </ogc:PropertyIsEqualTo>
+            <ogc:BBOX>
+                <ogc:PropertyName>SHAPE_GEOMETRY</ogc:PropertyName>
+                <gml:Envelope xmlns:gml=\"http://www.opengis.net/gml\" srsName=\"$bbox->[4]\">
+                    <gml:lowerCorner>$bbox->[0] $bbox->[1]</gml:lowerCorner>
+                    <gml:upperCorner>$bbox->[2] $bbox->[3]</gml:upperCorner>
+                </gml:Envelope>
+            </ogc:BBOX>
+        </ogc:And>
+    </ogc:Filter>";
+    $filter =~ s/\n\s+//g;
+
+    my $uri = URI->new("https://tilma.mysociety.org/proxy/occ/nsg/");
+    $uri->query_form(
+        REQUEST => "GetFeature",
+        SERVICE => "WFS",
+        SRSNAME => "urn:ogc:def:crs:EPSG::4326",
+        TYPENAME => "WFS_DEFECTS_FOR_QUERYING",
+        VERSION => "1.1.0",
+        filter => $filter,
+        propertyName => 'ITEM_CATEGORY_NAME,ITEM_TYPE_NAME,REQUIRED_COMPLETION_DATE,SHAPE_GEOMETRY',
+        outputformat => "application/json"
+    );
+
+    try {
+        my $response = get($uri);
+        my $json = JSON->new->utf8->allow_nonref;
+        return $json->decode($response);
+    } catch {
+        # Ignore WFS errors.
+        return [];
```

Comment:
`pins_from_wfs` is expecting a hashref, so this change would prevent a crash on L365 if the WFS response isn't valid JSON:

```suggestion
        return {};
```

Reply:
Oops! Fixed in c0b845fd55d5772003030ae374f82de8c69c9899